pycbc/waveform/bank.py

```diff
@@ -394,24 +395,53 @@ def approximant(self, index):
     def __len__(self):
         return len(self.table)
 
-    def template_thinning(self, injection_parameters, threshold):
-        from pycbc.pnutils import mass1_mass2_to_tau0_tau3
+    def generate_custom_size_waveform(self, t_num, max_freq, delta_f,
+                                      low_freq_cutoff=None, cached_mem=None):
```

Comment:
Changing this to generate_with_delta_f_and_max_freq and changing low_freq_cutoff name.